rustls/src/msgs/persist.rs

```diff
@@ -245,6 +311,30 @@ impl ClientSessionCommon {
         }
     }
 
+    /// [`Codec::read()`] is inherent here to avoid leaking the [`Codec`]
+    /// implementation through [`Deref`] implementations on
+    /// [`Tls12ClientSessionValue`] and [`Tls13ClientSessionValue`].
```

Comment:
Double checking my own understanding: it would be tempting to `impl Codec for ClientSessionCommon` but since `Tls13ClientSessionValue` and `Tls12ClientSessionValue` deref to `ClientSessionCommon` we'd be leaking the `Codec` trait. Instead similar methods are directly imp'd on the concrete types and in the future we may want to clean up the `ServerSessionValue` type similarly (since it presently `impl`'s `Codec` directly).

Is that a fair summary or am I off the mark?